### ISSUE TITLE
DynamicDiscription strips out accented characters

### ISSUE THREAD

**Issue by risingisland** [Nicola1971/Seo4Evo#8]
I am having a problem with this not displaying any accented characters, áéíóúñ etc.
If the page content starts out like "Bla bla áéíóúñ more bla bla", the output will be "Bla bla more bla bla"

Is this something you are willing to help out with?

**Comment by risingisland** [comment]
If I just copy and paste "Bla bla áéíóúñ more bla bla" into the meta description field, it displays as expected.

**Comment by risingisland** [comment]
**Update:** It has to do with line 130
`$text = preg_replace('~&.+;~U',' ',$text);`
which is commented "remove special MODx tags - chunks, snippets, etc."
But seems to be removing more.

**Comment by kulmjens** [comment]
I have inserted the following above line 130:
```$text = html_entity_decode($text,ENT_HTML5,'UTF-8');```
and now I have all my umlauts an accents back again. It converts something like `&auml;` to `ä`.
Maybe @Nicola1971 could revise this if the next line breaks when there are "special MODx tags - chunks, snippets, etc." in it. :-)

Another workaround would be to have TinyMCE put out the original letter instead of the HTML-entity.

**Comment by Nicola1971** [comment]
sorry for late reply
@kulmjens your solution seems working fine in my test.

**Comment by Nicola1971** [comment]
added one more line to remove some empty spaces after html_entity_decode
https://github.com/Nicola1971/Seo4Evo/blob/master/install/assets/snippets/DynamicDescription.tpl#L137

**Comment by risingisland** [comment]
I dont know if it makes any difference, but another solution that I found was to change line 130 from:
`*$text = preg_replace('~&.+;~U',' ',$text);`
to
`$text = preg_replace('~&+;~U',' ',$text);`

**Comment by Nicola1971** [comment]
@risingisland thanks I will try tomorrow.
Please check if lasted modifications works with your language.

**Comment by kulmjens** [comment]
@risingisland in my understanding of regexp, the difference between `&.+;` and `&+;` is, that the first regexp matches everything that starts with an `&`, followed by at least 1 character and ends with `;` (like `&auml;` or `&xyz;`, whereas the last regexp matches everything that has one or many `&` followed by a `;` (like `&;` or `&&&&;`). Of course it leaves all HTML entities alone. Correct me if I'm wrong, but this regexp doesn't make any sense to me.

**Comment by risingisland** [comment]
I had a hard time making sense of any of it   :blush:
So, if the `$text = html_entity_decode($text,ENT_HTML5,'UTF-8');` solution does the same job, it doesnt make any difference to me.

**Comment by kulmjens** [comment]
@risingisland :-)
The two lines don't do the same thing!
The line `preg_replace('~&.+;~U',' ',$text);` replaces every "word" that starts with an `&` and ends with `;` with a blank.
The line `html_entity_decode($text,ENT_HTML5,'UTF-8');` converts all known HTML entities into their normal text versions: like `&auml;` => ä , `&yen;` => ¥ or `&eacute;` => é. But it leaves something like `&abcd;` or `&xxx;` alone.
What I wanted to point out was, that in the regexp `&.+;` the plus belongs to the period and means "any character, at least one of it", while in the regexp `&+;` the plus belongs to the `&` and means "the character '&', at least one of it'.
So the last regexp only replaces `&;`, `&&;`, `&&&;`, `&&&&;`  (and so on) with a blank and that doesn't make any sense.
@Nicola1971 I added the flag `ENT_QUOTES` to the `html_entity_decode` to convert also double and single quotes. After that I added a line that deletes double and single quotes, as they break the meta tag parameter. So my suggestion would be:
```
$text = html_entity_decode($text,ENT_QUOTES|ENT_HTML5,'UTF-8');
$text = str_replace(array('"',"'"),'',$text);
```

**Comment by Nicola1971** [comment]
@kulmjens added your fix to 1.0.2 thanks
